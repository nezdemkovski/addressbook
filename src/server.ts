import bodyParser from 'body-parser';
import debug from 'debug';
import express from 'express';
import logger from 'morgan';

import { PORT } from './config';
import db from './db';
import indexRouter from './routes/index';
import usersRouter from './routes/users';

db();
debug('strv:server');
const app: express.Application = express();

app.use(logger('dev'));
app.use(bodyParser.json());

app.use('/', indexRouter);
app.use('/users', usersRouter);

app.listen(PORT, () => {
  /* tslint:disable-next-line:no-console */
  console.log(`Listening at http://localhost:${PORT}/`);
});
