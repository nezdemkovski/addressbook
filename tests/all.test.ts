import request from 'supertest';

import server from '../src/index';

process.env.NODE_ENV = 'test';

describe('GET /random-url', () => {
  it('should return 404', done => {
    request(server)
      .get('/random-url')
      .expect(404, done);
  });
});

describe('Home route tests', () => {
  test('get home route GET /', async () => {
    const response = await request(server).get('/');
    expect(response.status).toEqual(200);
    expect(response.text).toMatchSnapshot();
  });
});
