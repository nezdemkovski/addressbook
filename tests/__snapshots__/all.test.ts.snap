// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home route tests get home route GET / 1`] = `"{\\"data\\":\\"Hello from Yuri!\\"}"`;
